cli/src/main/java/org/projectnessie/tools/catalog/migration/cli/BaseRegisterCommand.java

```diff
@@ -0,0 +1,337 @@
+            + "If not specified, uses the present working directory.",
+        "Example: --output-dir /tmp/output/",
+        "         --output-dir $PWD/output_folder"
+      })
+  private Path outputDirPath;
+
+  @CommandLine.Option(
+      names = {"--dry-run"},
+      description =
+          "Optional configuration to simulate the registration without actually registering. Can learn about a list "
+              + "of tables that will be registered by running this.")
+  private boolean isDryRun;
+
+  @CommandLine.Option(
+      names = {"--disable-safety-prompts"},
+      description = "Optional configuration to disable safety prompts which needs console input.")
+  private boolean disablePrompts;
+
+  @CommandLine.Option(
+      names = {"--stacktrace"},
+      description =
+          "Optional configuration to enable capturing stacktrace in logs in case of failures.")
+  private boolean enableStackTrace;
+
+  private static final int BATCH_SIZE = 100;
+  public static final String FAILED_IDENTIFIERS_FILE = "failed_identifiers.txt";
+  public static final String FAILED_TO_DELETE_AT_SOURCE_FILE = "failed_to_delete_at_source.txt";
+  public static final String DRY_RUN_FILE = "dry_run_identifiers.txt";
+
+  private static final Logger consoleLog = LoggerFactory.getLogger("console-log");
+
+  public BaseRegisterCommand() {}
+
+  protected abstract CatalogMigrator catalogMigrator(
+      Catalog sourceCatalog, Catalog targetCatalog, boolean enableStackTrace);
+
+  protected abstract boolean canProceed(Catalog sourceCatalog);
+
+  protected abstract String operation();
+
+  protected abstract String operated();
+
+  protected abstract String operate();
+
+  @Override
+  public Integer call() {
+    Set<TableIdentifier> identifiers = Collections.emptySet();
+    String identifierRegEx = identifierOptions != null ? identifierOptions.identifiersRegEx : null;
+
+    if (identifierOptions != null) {
+      identifiers = identifierOptions.processIdentifiersInput();
+    }
+    checkAndWarnAboutIdentifiers(identifiers, identifierRegEx);
+
+    validateOutputDir();
+
+    Catalog sourceCatalog = sourceCatalogOptions.build();
+    consoleLog.info("Configured source catalog: {}", sourceCatalog.name());
+
+    Catalog targetCatalog = targetCatalogOptions.build();
+    consoleLog.info("Configured target catalog: {}", targetCatalog.name());
+
+    if (!isDryRun && !disablePrompts && !canProceed(sourceCatalog)) {
+      return 2;
+    }
+
+    CatalogMigrator catalogMigrator =
+        catalogMigrator(sourceCatalog, targetCatalog, enableStackTrace);
+
+    if (identifiers.isEmpty()) {
+      consoleLog.info("Identifying tables for {} ...", operation());
+      identifiers = catalogMigrator.getMatchingTableIdentifiers(identifierRegEx);
+      if (identifiers.isEmpty()) {
+        consoleLog.warn(
+            "No tables were identified for {}. Please check `catalog_migration.log` file for more info.",
+            operation());
+        return 2;
```

Comment:
`2` is a _reserved_ exit code and must not be used by applications

Reply:
`exit code 2` was used in 3 places. 
I have updated it with `exit code 1`

I also checked about using `IExitCodeExceptionMapper`, but I didn't have any special exception to throw for these 3 cases. Hence, didn't use.